presto-main/src/main/java/com/facebook/presto/operator/index/IndexLoader.java

```diff
@@ -334,7 +350,15 @@ public boolean load(List<UpdateRequest> requests)
                 driver.updateSource(new TaskSource(sourcePlanNodeId, ImmutableSet.of(split), true));
                 while (!driver.isFinished()) {
                     ListenableFuture<?> process = driver.process();
-                    checkState(process.isDone(), "Driver should never block");
+                    try {
+                        process.get(timeout.toMillis(), MILLISECONDS);
+                    }
+                    catch (TimeoutException e) {
+                        throw new PrestoException(INDEX_LOADER_TIMEOUT, format("Exceeded the time limit of %s loading indexes for index join", timeout));
+                    }
+                    catch (InterruptedException | ExecutionException e) {
```

Comment:
`InterruptedException` should interrupt current thread. Also, it might be better we unwrap the `ExecutionException` to get more meaningful stack trace but that's should be minor difference.

Reply:
I'm not sure what you mean about the Interrupted Exception.  The current thread would probably get interrupted too, but we have to catch the InterruptedException here anyway because otherwise we'd have an uncaught exception.

Reply:
It is always a safe play to interrupt the thread when there is a `InterruptedException` unless you know for sure we can ignore it:
https://www.ibm.com/developerworks/library/j-jtp05236/

Reply:
I see, thanks for the link.  So throwing a PrestoException isn't sufficiently interruptive because it can be caught without realizing it was really an InterruptedException.  So instead I should explicitly interrupt the current thread.